mmdeploy/codebase/mmdet/models/dense_heads/solo_head.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Dict, List
+
+import torch
+from mmdet.models.layers import mask_matrix_nms
+from torch import Tensor
+from torch.nn import functional as F
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    'mmdet.models.dense_heads.SOLOHead.predict_by_feat')
+def solohead__predict_by_feat__default(ctx, self,
+                                       mlvl_mask_preds: List[Tensor],
+                                       mlvl_cls_scores: List[Tensor],
+                                       batch_img_metas: List[Dict], **kwargs):
+    """Rewrite `predict_by_feat` of `SOLOHead` for default backend."""
+
+    def empty_results(cls_scores, ori_shape):
+        batch_size = cls_scores.size(1)
+        scores = cls_scores.new_ones(batch_size, 0, 1)
+        masks = cls_scores.new_zeros(batch_size, 0, *ori_shape)
+        labels = cls_scores.new_ones(batch_size, 0)
+        bboxes = cls_scores.new_zeros(batch_size, 0, 4)
+        dets = torch.cat([bboxes, scores], -1)
+        return dets, labels, masks
+
+    batch_size = mlvl_cls_scores[0].size(0)
+    cfg = self.test_cfg
+    mlvl_cls_scores = [
+        item.permute(0, 2, 3, 1).view(batch_size, -1, self.cls_out_channels)
+        for item in mlvl_cls_scores
+    ]
+    assert len(mlvl_mask_preds) == len(mlvl_cls_scores)
+    batch_mlvl_cls_scores = torch.cat(mlvl_cls_scores, dim=1)
+    batch_mlvl_mask_preds = torch.cat(mlvl_mask_preds, dim=1)
+    featmap_size = batch_mlvl_mask_preds.size()[-2:]
+    h, w = batch_img_metas[0]['img_shape'][:2]
+    upsampled_size = (featmap_size[0] * 4, featmap_size[1] * 4)
+    cls_labels = batch_mlvl_cls_scores.argmax(dim=-1)
+    batch_mlvl_cls_scores = torch.max(batch_mlvl_cls_scores, -1).values
+    score_mask = (batch_mlvl_cls_scores > cfg.score_thr)
+    batch_mlvl_cls_scores = batch_mlvl_cls_scores.where(
+        score_mask, batch_mlvl_cls_scores.new_zeros(1)).view(-1)
+    if len(batch_mlvl_cls_scores) == 0:
```

Comment:
It is a bad idea to apply branch inside a rewriter, the trace will not get the expected result.